### ISSUE TITLE
[java] AvoidReassigningParameters reports violations on wrong line numbers

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3254]
**Affects PMD Version:** 6.34.0

**Description:**

The rule [AvoidReassigningParameters](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#avoidreassigningparameters) reports each assignment. But the line reported is the declaration of the parameter instead of the assignment.

This is not helpful. Consider the example below with one parameter. Either only one violation should be reported at the declaration of the parameter or two violations on the two lines of the assignments.

**Code Sample demonstrating the issue:**

```java
public class AvoidReassigningParameters {
    public void a(String b) {  // line 2: here are two violations reported
        b = "1";               // line 3: no violation, but expected
        b = "2";               // line 4: no violation, but expected
    }
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
